Error ------------------------------------------------------------------------------------------------------ test.js:8:5

`Promise` in async scope is unused. Did you mean to `await` it? [unused-promise]

   8|     foo(); // error
          ^^^^^^


Error ----------------------------------------------------------------------------------------------------- test.js:10:5

`Promise` in async scope is unused. Did you mean to `await` it? [unused-promise]

   10|     bar(); // error
           ^^^^^^


Error ----------------------------------------------------------------------------------------------------- test.js:31:5

`Promise` in sync scope is unused. Promises must be handled by calling .then with a rejection handler, .catch, or
.finally. [unused-promise]

   31|     foo(); // error
           ^^^^^^


Error ----------------------------------------------------------------------------------------------------- test.js:32:5

`Promise` in sync scope is unused. Promises must be handled by calling .then with a rejection handler, .catch, or
.finally. [unused-promise]

   32|     foo().then(() => {}); // error
           ^^^^^^^^^^^^^^^^^^^^^


Error ----------------------------------------------------------------------------------------------------- test.js:33:5

`Promise` in sync scope is unused. Promises must be handled by calling .then with a rejection handler, .catch, or
.finally. [unused-promise]

   33|     foo().then(() => {}).then(() => {}); // error
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^


Error ----------------------------------------------------------------------------------------------------- test.js:37:5

`Promise` in sync scope is unused. Promises must be handled by calling .then with a rejection handler, .catch, or
.finally. [unused-promise]

   37|     b && foo(); // error
           ^^^^^^^^^^^


Error ----------------------------------------------------------------------------------------------------- test.js:38:5

`Promise` in sync scope is unused. Promises must be handled by calling .then with a rejection handler, .catch, or
.finally. [unused-promise]

   38|     b && foo() && b; // error
           ^^^^^^^^^^^^^^^^


Error ----------------------------------------------------------------------------------------------------- test.js:39:5

`Promise` in sync scope is unused. Promises must be handled by calling .then with a rejection handler, .catch, or
.finally. [unused-promise]

   39|     b && b && foo(); // error
           ^^^^^^^^^^^^^^^^


Error ----------------------------------------------------------------------------------------------------- test.js:40:5

`Promise` in sync scope is unused. Promises must be handled by calling .then with a rejection handler, .catch, or
.finally. [unused-promise]

   40|     foo() && foo(); // error
           ^^^^^^^^^^^^^^^


Error ----------------------------------------------------------------------------------------------------- test.js:41:5

`Promise` in sync scope is unused. Promises must be handled by calling .then with a rejection handler, .catch, or
.finally. [unused-promise]

   41|     foo().catch(() => {}) && foo(); // error
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^


Error ----------------------------------------------------------------------------------------------------- test.js:42:5

`Promise` in sync scope is unused. Promises must be handled by calling .then with a rejection handler, .catch, or
.finally. [unused-promise]

   42|     foo() && foo().catch(() => {}); // error
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^


Error ----------------------------------------------------------------------------------------------------- test.js:51:5

`Promise` in sync scope is unused. Promises must be handled by calling .then with a rejection handler, .catch, or
.finally. [unused-promise]

   51|     b ? foo() : 3; // error
           ^^^^^^^^^^^^^^


Error ----------------------------------------------------------------------------------------------------- test.js:52:5

`Promise` in sync scope is unused. Promises must be handled by calling .then with a rejection handler, .catch, or
.finally. [unused-promise]

   52|     b ? 3 : foo(); // error
           ^^^^^^^^^^^^^^


Error ----------------------------------------------------------------------------------------------------- test.js:65:5

`Promise` in sync scope is unused. Promises must be handled by calling .then with a rejection handler, .catch, or
.finally. [unused-promise]

   65|     x.foo(); // error
           ^^^^^^^^


Error ----------------------------------------------------------------------------------------------------- test.js:66:5

`Promise` in sync scope is unused. Promises must be handled by calling .then with a rejection handler, .catch, or
.finally. [unused-promise]

   66|     x.bar(); // error
           ^^^^^^^^


Error ----------------------------------------------------------------------------------------------------- test.js:80:5

`Promise` in sync scope is unused. Promises must be handled by calling .then with a rejection handler, .catch, or
.finally. [unused-promise]

   80|     x?.foo(); // error
           ^^^^^^^^^


Error ----------------------------------------------------------------------------------------------------- test.js:81:5

`Promise` in sync scope is unused. Promises must be handled by calling .then with a rejection handler, .catch, or
.finally. [unused-promise]

   81|     x?.bar(); // error
           ^^^^^^^^^



Found 17 errors
